scylla.go

```diff
@@ -189,12 +194,28 @@ func (p *scyllaConnPicker) Put(conn *Conn) {
 		copy(p.conns, conns)
 	}
 	if c := p.conns[s.shard]; c != nil {
-		conn.Close()
+		p.excessConns = append(p.excessConns, conn)
```

Comment:
Could we put a limit to how big `p.excessConns` could be, so that it does not grow unbounded if something goes wrong? Not sure how many connections we need to hit all shards, so I guess if we reach `10*p.nrShards` something is off and maybe we should close the connections anyway.

Reply:
I am not sure it helps. It can even make it worse. The idea is that the chance to get a new shard increases significantly with the number of connections so equilibrium should be reached faster.
What do you think @avikiviti ?

Reply:
You could have multiple clients connecting to the same host though, which would cause it to race, so theoretically you could keep opening connections forever (even if the chance drops with time), so I think it's still good to have some kind of limit.